mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -736,15 +765,27 @@ class GeneralizedLinearRegressionModel private[ml] (
   private lazy val familyAndLink = new FamilyAndLink(familyObj, linkObj)
 
   override protected def predict(features: Vector): Double = {
-    val eta = predictLink(features)
```

Comment:
it looks like if the family = Gaussian and link = Identity you don't even need to check for offsetCol here